### PR TITLE
Add support for dark_icon and dark_logo

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Allow add-on authors to add a  `dark_[icon|logo].png` file that will be requested when the user has dark mode enabled in the frontend.

If that file does not exist it will fallback to the regular `[icon|logo].png` file

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/1359
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by frenck — APPROVED** [review]
LGTM, nice :)

**Comment by pvizeli** [comment]
Have to wait until https://github.com/home-assistant/supervisor/pull/3623 is merged because the endpoints moved

**Comment by ludeeus** [comment]
Not usable without a core and a frontend change anyway

File: supervisor/api/__init__.py [hunk 1]
```diff
@@ -379,6 +379,10 @@ def _register_addons(self) -> None:
             [
                 web.get("/addons", api_addons.list),
                 web.get("/addons/{addon}/info", api_addons.info),
+                web.get("/addons/{addon}/icon", api_addons.icon),
+                web.get("/addons/{addon}/dark_icon", api_addons.dark_icon),
+                web.get("/addons/{addon}/logo", api_addons.logo),
+                web.get("/addons/{addon}/dark_logo", api_addons.dark_logo),
```
Comment (pvizeli) [inline supervisor/api/__init__.py:385]: move that into legacy part on store

Comment (ludeeus) [reply supervisor/api/__init__.py:385]: no?
/addons/* are used for installed addons
/store/addons/* are used for not installed

File: supervisor/api/addons.py [hunk 2]
```diff
@@ -412,6 +412,54 @@ def logs(self, request: web.Request) -> Awaitable[bytes]:
...
+        if not addon.with_logo:
+            raise APIError(f"No logo found for add-on {addon.slug}!")
+
+        if addon.with_dark_logo:
+            with addon.path_dark_logo.open("rb") as png:
+                return png.read()
+
+        with addon.path_logo.open("rb") as png:
+            return png.read()
+
```
Comment (pvizeli) [inline supervisor/api/addons.py:462]: We should readd this if we change the logic and cache it after installation to keep it same also if the store change it until next update. For now, we can just route over the legacy part until we add such logick

File: supervisor/api/__init__.py (hunk 1 above)
Comment (pvizeli) [reply supervisor/api/__init__.py:385]: see comment above

File: supervisor/api/addons.py (hunk 2 above)
Comment (ludeeus) [reply supervisor/api/addons.py:462]: See response above.
Adding new endpoints to "legacy" is not right, especially here where they _should_ have a different meaning.

File: supervisor/api/__init__.py (hunk 1 above)
Comment (mdegat01) [reply supervisor/api/__init__.py:385]: @pvizeli I'm confused too, why would this be legacy? Installed addons can have a dark icon/logo too

**Comment by frenck** [comment]
Sad to see this one closed to be honest :(

**Pull request closed without merging**